tools/ci-operator-prowgen/main.go

```diff
@@ -0,0 +1,126 @@
+package main
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"github.com/ghodss/yaml"
+	cioperatorapi "github.com/openshift/ci-operator/pkg/api"
+	"io/ioutil"
+	prowconfig "k8s.io/test-infra/prow/config"
+	prowkube "k8s.io/test-infra/prow/kube"
+	"os"
+	"path"
+	"strings"
+)
+
+type options struct {
+	ciOperatorConfigPath string
+
+	help    bool
+	verbose bool
+}
+
+func bindOptions(flag *flag.FlagSet) *options {
+	opt := &options{}
+
+	flag.StringVar(&opt.ciOperatorConfigPath, "source-config", "", "Path to ci-operator configuration file in openshift/release repository.")
+	flag.BoolVar(&opt.help, "h", false, "Show help for ci-operator-prowgen")
+	flag.BoolVar(&opt.verbose, "v", false, "Show verbose output")
+
+	return opt
+}
+
+func generatePodSpec() *prowkube.PodSpec {
```

Comment:
We are going to migrate to the "real" upstream `PodSpec` (from k8s api, not `test-infra`) soon. There is no reason other than historical silliness as to why the `test-infra` type is different. You should be OK to target the upstream type here and `test-infra` will catch up soon.